cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -49,142 +67,182 @@ describe('Unit test for ShapeData', () => {
+    cy.get('.notes').type(notes);
+    pressButtonAndWaitForPromise('save');
+    assertOnFirestoreAndPopup(path, withNotes(notes));
+  });
+
+  it('Deletes polygon', () => {
+    drawPolygonAndClickOnIt();
+    deletePolygon(cy.stub(window, 'confirm').returns(true));
+  });
+
+  it('Almost deletes, then deletes', () => {
+    drawPolygonAndClickOnIt();
+    let confirmValue = false;
+    const confirmStub =
+        cy.stub(window, 'confirm').callsFake(() => confirmValue);
+    pressPopupButton('delete').then(() => {
+      expect(confirmStub).to.be.calledOnce;
+      confirmStub.resetHistory();
     });
+    assertOnFirestoreAndPopup(path, defaultData);
+    setUpPage();
+    assertOnFirestoreAndPopup(path, defaultData);
+    pressPopupButton('delete').then(() => {
+      expect(confirmStub).to.be.calledOnce;
+      confirmValue = true;
+      confirmStub.resetHistory();
+    });
+    assertOnFirestoreAndPopup(path, defaultData);
+    deletePolygon(confirmStub);
   });
 
-  it('Delete shape', () => {
-    const popup = new StubPopup();
-    popup.setCalculatedData(calculatedData);
-    const underTest =
-        new StoredShapeData('my_id', 'my notes', polygonGeometry, popup);
-    popup.mapFeature.getMap = () => null;
-    const ids = [];
-    firebaseCollection.doc = (id) => {
-      ids.push(id);
-      return {
-        delete: () => Promise.resolve(),
-      };
-    };
-    cy.wrap(underTest.update()).then(() => {
-      expect(ids).to.eql(['my_id']);
-      expect(underTest.id).to.eql('my_id');
-      expect(StoredShapeData.pendingWriteCount).to.eql(0);
+  /**
+   * Presses a visible 'delete' button, checks that `confirmStub` was called,
+   * and that the polygon is gone and there is no data in Firestore.
+   * @param {Spy} confirmStub
+   */
+  function deletePolygon(confirmStub) {
+    pressPopupButton('delete').then(() => expect(confirmStub).to.be.calledOnce);
+    cy.get('#test-map-div').click();
+    cy.get('#test-map-div').should('not.contain', 'damage count');
+    getCurrentUpdatePromiseInCypress()
+        .then(() => userShapes.get())
+        .then((querySnapshot) => {
+          expect(querySnapshot).to.have.property('size', 0);
+          expect(querySnapshot.docs).to.be.empty;
+        });
+  }
+
+  it('Draws a polygon, clicks it, closes its info box', () => {
+    drawPolygonAndClickOnIt();
+    pressPopupButton('close');
+    cy.get('#test-map-div').contains('damage count');
+    cy.get('#test-map-div').contains('damage count').should('not.be.visible');
+  });
+
+  it('Draws a polygon, almost closes while editing', () => {
+    const confirmStub = cy.stub(window, 'confirm').returns(false);
+    drawPolygonAndClickOnIt();
+    pressPopupButton('edit');
+    cy.get('.notes').type(notes);
+    pressPopupButton('close').then(() => expect(confirmStub).to.be.calledOnce);
+    pressPopupButton('save');
+    cy.get('#test-map-div').contains(notes).should('be.visible');
+    getCurrentUpdatePromiseInCypress();
```

Comment:
calling this without doing anything with what it returns will just ensure that the promise is finished before executing more cypress things correct?